File: src/main/java/org/jukeboxmc/event/player/PlayerExperienceChangeEvent.java
package org.jukeboxmc.event.player;

import org.jukeboxmc.event.Cancellable;
import org.jukeboxmc.player.Player;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class PlayerExperienceChangeEvent extends PlayerEvent implements Cancellable {

    private final int oldExperience;
    private final int oldLevel;

    private int newExperience;
    private int newLevel;


    /**
     * Creates a new {@link PlayerEvent}
     *
     * @param player who represents the player which comes with this event
     */
    public PlayerExperienceChangeEvent( Player player, int oldExperience, int oldLevel, int newExperience, int newLevel ) {
        super( player );
        this.oldExperience = oldExperience;
        this.oldLevel = oldLevel;
        this.newExperience = newExperience;
        this.newLevel = newExperience;
    }

    public int getOldExperience() {
        return this.oldExperience;
    }

    public int getOldLevel() {
        return this.oldLevel;
    }

    public int getNewExperience() {
        return this.newExperience;
    }

    public void setNewExperience( int newExperience ) {
        this.newExperience = newExperience;
    }

    public int getNewLevel() {
        return this.newLevel;
    }

    public void setNewLevel( int newLevel ) {
        this.newLevel = newLevel;
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/player/PlayerExperienceChangeEvent.kt
package org.jukeboxmc.api.event.player

import org.jukeboxmc.api.event.Cancellable
import org.jukeboxmc.api.player.Player

class PlayerExperienceChangeEvent(
    player: Player,
    private val oldExperience: Float,
    private val oldLevel: Int,
    private var newExperience: Float,
    private var newLevel: Int
) : PlayerEvent(player), Cancellable{

    fun getOldExperience(): Float {
        return this.oldExperience
    }

    fun getOldLevel(): Int {
        return this.oldLevel
    }

    fun getNewExperience(): Float {
        return this.newExperience
    }

    fun setNewExperience(newExperience: Float) {
        this.newExperience = newExperience
    }

    fun getNewLevel(): Int {
        return this.newLevel
    }

    fun setNewLevel(newLevel: Int) {
        this.newLevel = newLevel
    }
}